rust/krb5-provision-keytab/src/main.rs

```diff
@@ -67,12 +69,20 @@ enum Error {
     RemoveDummyFromKeytab { source: krb5::Error },
 }
 
+#[derive(clap::Parser)]
+pub struct Args {
+    #[command(flatten)]
+    pub cluster_info_opts: KubernetesClusterInfoOpts,
```

Comment:
This will forward args set via envvar, but not ones passed over the command line. I *think* I would rather just hard-code a value here since it's not used by krb5-provision-keytab anyway.

Maybe it's a sign that KPK should just depend on kube directly, rather than via op-rs.

Reply:
That's actually what I had before and "improved" it in 547b5d2dda57c8cc0b6242831017cf0429c81305.
I knew that this only works for envs for now and was planning of adding CLI support today.
```rust
    let mut args = vec![];
    if let Some(kubernetes_cluster_domain) = &cluster_info_opts.kubernetes_cluster_domain {
        args.push(format!(
            "--kubernetes-cluster-domain {kubernetes_cluster_domain}"
        ))
    }

    let mut child = Command::new("stackable-krb5-provision-keytab")
        .args(args)
```

What do you think about this solution? We don't know if the stackable client will actually use the passed values, so I don't like hiding the actual config from it :/

Anyway, I'll quickly spike switching from op-rs to kube

Reply:
Ok, well that's a deep rabbit whole, let's stick to the stackable client

Reply:
99d6e525185250f956e21059606c4c1f669c4d6e should now correctly propagate the cluster domain to the command invocation

Reply:
Or at the very least, `let KubernetesClusterInfo { kubernetes_cluster_domain } = cluster_info;` so that any new fields added trigger a compilation error instead of being silently ignored.